dotnet/samples/KernelSyntaxExamples/Getting_Started/Step7_Observability.cs

```diff
@@ -15,10 +15,39 @@ namespace GettingStarted;
 
 public class Step7_Observability : BaseTest
 {
+    /// <summary>
+    /// Shows how to observe the execution of a <see cref="KernelPlugin"/> instance with filters.
+    /// </summary>
+    [Fact]
+    public async Task ObservabilityWithFiltersAsync()
+    {
+        // Create a kernel with OpenAI chat completion
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.AddOpenAIChatCompletion(
+                modelId: TestConfiguration.OpenAI.ChatModelId,
+                apiKey: TestConfiguration.OpenAI.ApiKey);
+
+        kernelBuilder.Plugins.AddFromType<TimeInformation>();
+
+        // Add filter using DI
+        kernelBuilder.Services.AddSingleton<ITestOutputHelper>(this.Output);
+        kernelBuilder.Services.AddSingleton<IFunctionFilter, MyFunctionFilter>();
+
+        Kernel kernel = kernelBuilder.Build();
+
+        // Add filter without DI
+        kernel.PromptFilters.Add(new MyPromptFilter(this.Output));
+
+        // Invoke the kernel with a prompt and allow the AI to automatically invoke functions
+        OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+        WriteLine(await kernel.InvokePromptAsync("How many days until Christmas? Explain your thinking.", new(settings)));
+    }
+
     /// <summary>
     /// Shows how to observe the execution of a <see cref="KernelPlugin"/> instance with hooks.
     /// </summary>
     [Fact]
+    [Obsolete("Events are deprecated in favor of filters.")]
```

Comment:
Why don't we just remove this from the sample to avoid confusion

Reply:
@markwallace-microsoft My idea was to keep sample with events at least for the short period of time after release so developers can see before/after approaches, compare both of them and observe the differences. But I can remove it if it's not going to be useful, please let me know.